streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1776,22 +1780,28 @@ private long advanceNowAndComputeLatency() {
         return Math.max(now - previous, 0);
     }
 
+    public void shutdown() {
```

Comment:
as mentioned above, let's just get rid of the overload without a `leaveGroup` parameter, this way the caller has to consciously decide whether it should be leaving the group or not

Reply:
Make sense!